### ISSUE TITLE
 PicoEMP pinout doubts 

### ISSUE THREAD

**Issue by ic32k** [newaetech/chipshouter-picoemp#32]
Hello,

I was looking for some more info abut the pinout of the PicoEMP. and what could be his possible usage.... 

I saw the introduction video, thus also tried to find other videos to see how they use it but there is no may out there, also tried to look for info in the ‘user manual’ of ChipShouter, but I was’t lucky enough to find the usage of the pins, so here are my questions

It has 3 pin headers:
P1: GP0 to GP5 and GND
P2: HVP and GND
P3: CHG, VCC, GND and PWM

The readme says “External HVP mode: a pulse generator can be used to control EM pulse insertion”, so I understand it must be connected to P2 header and I can attach to it any 'clock source', but how we control when the pulse is done with this mode?

Also says “Fast triggering using GPIO0” this is GP0 from pin header P1, isn’t it? It can be used to connect GPIO4 from target board and trigger a pulse when rising edge is received like with chipwhisperer? or maybe that fast triggering is taking GP0 to VCC/GND?

What are header P3’s “CHG” and “PWM” pins for?

Also wanted to know if the pulse power value of the source code (#define PULSE_POWER_DEFAULT 0.0122) has direct relation with the coil tip characteristics, and if so how can we make calculations to keep it under safe values? IDK maybe there is a min or max level can be configured based on uH value of the coil like let’s say I have a coil with 0.8mm wire, 4mm diameter 15 windings and 3.3uH so calculating X Z and Y the safe or valid power values are from 0.0050 to 1.3259 (This values are completely fictitious)

And the last question... What this GP1 used for? GP2 to GP5 has any use?

Thank for your help guys!!!

**Comment by gquere** [comment]
> Also says “Fast triggering using GPIO0” this is GP0 from pin header P1, isn’t it?

Yes, do note that you have to re-enable fast triggering each time it's been externally triggered.
